FirebaseRemoteConfig/Tests/Unit/RCNInstanceIDTest.m

```diff
@@ -0,0 +1,400 @@
+                                                                   namespace:fullyQualifiedNamespace
+                                                                     options:currentOptions]);
+  }
+}
+
+// Mock instance ID methods.
+- (void)mockInstanceIDMethodForTokenAndIdentity:(nullable NSString *)token
+                                     tokenError:(nullable NSError *)tokenError
+                                       identity:(nullable NSString *)identity
+                                  identityError:(nullable NSError *)identityError {
+  // Mock the instanceID retreival method.
+  id instanceIDMock = OCMPartialMock([FIRInstanceID instanceIDForTests]);
+  OCMStub([instanceIDMock
+      tokenWithAuthorizedEntity:[OCMArg any]
+                          scope:[OCMArg any]
+                        options:[OCMArg any]
+                        handler:([OCMArg
+                                    invokeBlockWithArgs:(token ? token : [NSNull null]),
+                                                        (tokenError ? tokenError : [NSNull null]),
+                                                        nil])]);
+  OCMStub([instanceIDMock
+      getIDWithHandler:([OCMArg invokeBlockWithArgs:(identity ? identity : [NSNull null]),
+                                                    (identityError ? identityError : [NSNull null]),
+                                                    nil])]);
+}
+
+- (void)tearDown {
+  [_DBManager removeDatabaseOnDatabaseQueueAtPath:_DBPath];
+  [[NSUserDefaults standardUserDefaults] removePersistentDomainForName:_userDefaultsSuiteName];
+  [super tearDown];
+}
+
+// Instance ID token is nil. Error is not nil. Verify fetch fails.
+- (void)testNilInstanceIDTokenAndError {
+  NSMutableArray<XCTestExpectation *> *expectations =
+      [[NSMutableArray alloc] initWithCapacity:RCNTestRCNumTotalInstances];
+
+  // Test for each RC FIRApp, namespace instance.
+  for (int i = 0; i < RCNTestRCNumTotalInstances; i++) {
+    // Set the token as nil.
+    [self mockInstanceIDMethodForTokenAndIdentity:nil
+                                       tokenError:[NSError errorWithDomain:@"com.google.instanceid"
+                                                                      code:FIRInstanceIDErrorUnknown
+                                                                  userInfo:nil]
+                                         identity:nil
+                                    identityError:nil];
+
+    expectations[i] =
+        [self expectationWithDescription:
+                  [NSString stringWithFormat:@"Test fetch configs successfully - instance %d", i]];
+    XCTAssertEqual(_configInstances[i].lastFetchStatus, FIRRemoteConfigFetchStatusNoFetchYet);
+    FIRRemoteConfigFetchCompletion fetchCompletion =
+        ^void(FIRRemoteConfigFetchStatus status, NSError *error) {
+          XCTAssertNotNil(error);
+          [expectations[i] fulfill];
```

Comment:
Do we expect a particular `status` value here?

Reply:
Done.

Reply:
Would you like to add a particular `status` check to other similar tests?